### DISCUSSION TITLE
Restart helper applications after Sparkle update?

### DISCUSSION THREAD

**Discussion by latenitefilms** [sparkle-project/Sparkle#2621] [Q&A]
I have an application, let's say called `MyApp.app` within that application bundle in the `Contents/Helpers` folder is two other helper applications. When I update `MyApp.app` using Sparkle, does Sparkle automatically handle the closing of those Helper applications, or do I need to add some logic of my own to make that happen?

**Comment by latenitefilms** [comment]
It seems Sparkle doesn't automatically handle this.

I've solved by using:

```
func updaterWillRelaunchApplication(_ updater: SPUUpdater) {
```

**Comment by zorgiepoo** [comment]
Using updaterWillRelaunchApplication is not advised in this context and may not work in all cases.

You should handle this yourself when your app quits and is next launched, and not depend on Sparkle.

Users can update your app manually too, without Sparkle intervention. Sparkle can install updates on quit without user explicitly relaunching the app. Other app updaters could try to update your app. Etc.

XPC Service helpers, for example, terminate when the main app process terminates.
